File: server/FitMate.Core/JsonModels/ProgramPlans/ProgramTodayModel.cs
namespace FitMate.Core.JsonModels.ProgramPlans;

public class ProgramTodayModel
{
    public DateOnly Date { get; set; }
    public bool HasActiveProgram { get; set; }
    public long? ProgramId { get; set; }
    public string? ProgramName { get; set; }
    public ProgramPlanDayModel? Today { get; set; }
    public ProgramPlanDayModel? MissedWorkout { get; set; }
    public ProgramPlanDayModel? NextWorkout { get; set; }
}

File: client/src/types/JsonModels/ProgramPlans/ProgramTodayModel.ts
import type { JsonModels } from "../../backend";

export interface ProgramTodayModel
{
	date: string;
	hasActiveProgram: boolean;
	programId?: number;
	programName?: string;
	today?: JsonModels.ProgramPlans.ProgramPlanDayModel;
	missedWorkout?: JsonModels.ProgramPlans.ProgramPlanDayModel;
	nextWorkout?: JsonModels.ProgramPlans.ProgramPlanDayModel;
}
